substrate/frame/support/src/lib.rs

```diff
@@ -117,12 +117,27 @@ pub const LOG_TARGET: &str = "runtime::frame-support";
 #[derive(Encode, Decode, Debug, PartialEq, Eq, Clone, TypeInfo)]
 pub enum Never {}
 
+/// Utility function to declare string literals backed by an array of length N.
+/// The input can be shorter than its underlaying array in which case the end of the array is padded
+/// with zeros. It can be used in const contexts.
+pub const fn s<const N: usize>(s: &str) -> [u8; N] {
```

Comment:
Maybe we put this into `sp-runtime` as @ggwpez said. General I'm not happy with this position either. The naming should also be improved `s` is not really expressive.

Reply:
No problem, changing to `str_array`, I use it often in embedded projects to the point `s` becomes a "well-known identifier"

Reply:
Yeah I also have seen this before, but I like it to be more expressive. Ty!